Source/WebCore/dom/Document.cpp

```diff
@@ -4454,8 +4454,13 @@ void Document::runScrollSteps()
     if (frameView) {
         MonotonicTime now = MonotonicTime::now();
         bool scrollAnimationsInProgress = serviceScrollAnimationForScrollableArea(frameView.get(), now);
-        if (auto* scrollableAreas = frameView->scrollableAreas()) {
-            for (auto* scrollableArea : *scrollableAreas) {
+        if (frameView->scrollableAreas()) {
+            auto scrollableAreasToUpdate = *frameView->scrollableAreas();
+            if (auto nonUserScrollableAreas = frameView->scrollableAreasForAnimatedScroll()) {
+                for (auto* area : *nonUserScrollableAreas)
+                    scrollableAreasToUpdate.add(area);
```

Comment:
Can you do `scrollableAreasToUpdate.add(nonUserScrollableAreas.begin(), nonUserScrollableAreas.end())`?

Reply:
Fixed.